src/test/java/com/microsoft/graph/serializer/FallbackTypeAdapterFactoryTest.java

```diff
@@ -0,0 +1,54 @@
+package com.microsoft.graph.serializer;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.reflect.TypeToken;
+import com.microsoft.graph.http.BaseCollectionResponse;
+import com.microsoft.graph.logger.ILogger;
+import com.microsoft.graph.models.MessageStub;
+import com.microsoft.graph.models.MessagesCollectionResponseStub;
+import com.microsoft.graph.models.SubReactionStub1;
+import com.microsoft.graph.models.SubReactionStub2;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Type;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+
+public class FallbackTypeAdapterFactoryTest {
+
+    /**
+     * This test covers the scenario of objects not being
+     * properly deserialized (@odata.type not taken into account):
+     * https://github.com/microsoftgraph/msgraph-sdk-java-core/pull/249
+     */
+    @Test
+    public void testDeserializationOfNestedODataTypeAnnotatedObjects() {
+        final ILogger logger = mock(ILogger.class);
```

Comment:
this tests the case for collections. Could we add test cases for:
- property value of an object (i.e. a response which is a single message, not a collection of messages, and which has a reaction with the odata type set)
- property values of an object (same as above, but the property is a collection of objects this time)